### DISCUSSION TITLE
How to contribute to a github project using a fork

### DISCUSSION THREAD

**Discussion by thomasvt** [community/community#23174] [New to GitHub]
<ol>
<li>I contributed a bugfix to someone else’s github project a few days ago. Github kinda walked me through it to make a fork of the project on my own account, fix the bug there and do a pull request to the original project.</li>
<li>The project approved and committed my fix.</li>
<li>I pull requested the latest changes of that project to my own fork so I have their latest version. This caused a commit on my own fork containing that merge. Fine…</li>
<li>But now, when I want to fix another bug and pull request it to the original project, that commit from step (3) will be included in that PR, which makes no sense to their project.</li>
</ol>
<p>Am I missing something?</p>
<p>thank you</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="thomasvt" data-post="1" data-topic="159541">
<div class="title">
<div class="quote-controls"></div>
 thomasvt:</div>
<blockquote>
<p>I pull requested the latest changes of that project to my own fork so I have their latest version. This caused a commit on my own fork containing that merge. Fine…</p>
</blockquote>
</aside>
<p>That’s exactly what’s causing the problem. Usually you should fast-forward merge the upstream updates into the default branch of your fork. Unfortunately there’s no built-in tool to do that, you need to either do that locally and push to your fork, or set up an Actions workflow to do it.</p>
<p>To get back now, you could (again, locally) <code>git reset</code> the default branch and then force-push that to your fork. If you have already made further commits you might need to rebase those to preserve them.</p>
